### DISCUSSION TITLE
Noticeable input lag when using HDMI and USB[RPi Zero W]

### DISCUSSION THREAD

**Discussion by SynthGal** [probonopd/MiniDexed#733] [Q&A] [answered]
Not much I can add beyond the title. Issue happens with both a MIDI USB keyboard and a regular old USB typing keyboard. The lag on HDMI audio out is approximately 20ms based on my terrible attempts to measure with a stopwatch. Is there any way to reduce this latency?

(PS: thanks for allowing regular USB keyboards to work at all, I really appreciate this feature. I'd suggest mentioning it on the front page so people without MIDI keyboards know they can mess around with MiniDexed regardless.)

EDIT: I tried using MIDIDumpEnabled to see if there was any lag between the device getting a MIDI command and actually synthesizing the audio, but turns out this setting doesn't work on my RPi. 

EDIT 2: I plugged in my KORG nanoKontrol2, and turns out MIDIDumpEnabled works, but only with analog controls. Is there a reason note on commands are ignored?

**Comment by stanleygarvey** [comment, marked as answer]
You will always have lag on hdmi, it's not real-time. Use a dac or headphone out.

8 Oct 2024 22:05:52 SynthGal ***@***.***>:
